### PR TITLE
[python] Use bindings for `SparseNDArray` readpath

### PR DESCRIPTION
**Pull request by nguyenv**
**Issue and/or context:**

https://github.com/single-cell-data/TileDB-SOMA/issues/2077

**Changes:**

Use `clib.SparseNDArray` for read path, replacing usage of `tiledb.Array`.

These changes were picked out of https://github.com/single-cell-data/TileDB-SOMA/pull/1817.

**Notes for Reviewer:**

These changes are on top of https://github.com/single-cell-data/TileDB-SOMA/pull/2132.

### CODE REVIEW

**Comment by nguyenv** [comment]
Needs to be updated to use `SOMAContext` once #2158 is merged.

**Comment by nguyenv** [comment]
This is ready for review.

File: apis/python/src/tiledbsoma/_sparse_nd_array.py [hunk 1]
```diff
@@ -155,11 +155,27 @@ def read(
               ``slice(2,None)`` or ``slice(None,4)``.
             * Negative indexing is unsupported.
         """
-        del batch_size, platform_config  # Currently unused.
+        del batch_size  # Currently unused.
+        handle: clib.SparseNDArrayWrapper = self._handle._handle
```
Comment (thetorpedodog) [inline apis/python/src/tiledbsoma/_sparse_nd_array.py:159]: Does this type exist?

File: apis/python/src/tiledbsoma/soma_object.cc [hunk 2]
```diff
@@ -81,7 +81,6 @@ void load_soma_object(py::module& m) {
             "context"_a,
             py::kw_only(),
             "timestamp"_a = py::none())
-
         .def_property_readonly("type", &SOMAObject::type);
 };
-}  // namespace libtiledbsomacpp
+}  // namespace libtiledbsomacpp
```
Comment (thetorpedodog) [inline apis/python/src/tiledbsoma/soma_object.cc:86]: nit: lost end-of-file newline

**Review by thetorpedodog — APPROVED** [review]

**Comment by aaronwolen** [comment]
[sc-40220]

**Pull request merged**